airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,546 @@
+
+        :param secret_path: Path to read from
+        :type secret_path: str
+        :rtype: dict
+        :return: secret metadata. This is a Dict containing metadata for the secret.
+
+                 See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
+
+        """
+        return self.vault_client.get_secret_metadata(secret_path=secret_path)
+
+    def get_secret_including_metadata(self,
+                                      secret_path: str,
+                                      secret_version: Optional[int] = None) -> Optional[dict]:
+        """
+        Reads secret including metadata. It is only valid for KV version 2.
+
+        See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
+
+        :param secret_path: Path of the secret
+        :type secret_path: str
+        :param secret_version: Optional version of key to read - can only be used in case of version 2 of KV
+        :type secret_version: int
+        :rtype: dict
+        :return: key info. This is a Dict with "data" mapping keeping secret
+                 and "metadata" mapping keeping metadata of the secret.
+
+        """
+        return self.vault_client.get_secret_including_metadata(
+            secret_path=secret_path, secret_version=secret_version)
+
+    def create_or_update_secret(self,
+                                secret_path: str,
+                                secret: dict,
+                                method: Optional[str] = None,
+                                cas: Optional[int] = None) -> Response:
+        """
+        Creates or updates secret.
+
+        :param secret_path: Path to read from
+        :type secret_path: str
+        :param secret: Secret to create or update for the path specified
+        :type secret: dict
+        :param method: Optional parameter to explicitly request a POST (create) or PUT (update) request to
+            the selected kv secret engine. If no argument is provided for this parameter, hvac attempts to
+            intelligently determine which method is appropriate. Only valid for KV engine version 1
+        :type method: str
+        :param cas: Set the "cas" value to use a Check-And-Set operation. If not set the write will be
+            allowed. If set to 0 a write will only be allowed if the key doesn't exist.
+            If the index is non-zero the write will only be allowed if the key's current version
+            matches the version specified in the cas parameter. Only valid for KV engine version 2.
+        :type cas: int
+        :rtype: requests.Response
+        :return: The response of the create_or_update_secret request.
+
+                 See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v1.html
+                 and https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
```

Comment:
Same question here about how does this render.

Reply:
Will check.